Automattic/wp-calypso
# Earn: add one time payment option to payment plans

#### Changes proposed in this Pull Request

Exposes the one time payment option. Copy changes are handled in https://github.com/Automattic/wp-calypso/pull/42848 under the same feature flag

This also includes a fix for one time payments being used to back premium content.

<img width="823" alt="Screen Shot 2020-06-17 at 5 01 00 PM" src="https://user-images.githubusercontent.com/1270189/84962668-410e0700-b0bc-11ea-9185-cf602db6ab26.png">
<img width="862" alt="Screen Shot 2020-06-17 at 5 01 12 PM" src="https://user-images.githubusercontent.com/1270189/84962677-4408f780-b0bc-11ea-972f-845a32f2946c.png">

### Payments button testing instructions
* Set up a recurring payment plan on a site using the testing store. More info in PCYsg-lW4-p2 #sandbox-method.
* Checkout this branch locally, and verify that a new one-time payment option is available in development
* Verify no changes in production or add `?flags=-earn/rename-payment-blocks` and refresh
* Try adding recurring payments button

### Premium Content Testing Instructions
* (Requires a sandbox) Checkout this branch and either apply the linked diff, or run cd apps/full-site-editing && yarn dev --sync
* Set up a recurring payment plan on a site using the testing store. More info in PCYsg-lW4-p2 #sandbox-method.
* Insert a premium content block and select a one-time payment block
* With another user, pay for the premium content
* Verify that we can see this content without errors

Fixes https://github.com/Automattic/wp-calypso/issues/42858
